includes/class-wpcom-legacy-redirector-ui.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+class WPCOM_Legacy_Redirector_UI {
+
+    public function admin_menu() {
+        add_menu_page( 'WPCOM Legacy Redirector', 'Redirect Manager', 'manage_options', 'wpcom-legacy-redirector', array(
+            $this,
+            'generate_page_html',
+        ), 'dashicons-external' );
+    }
+
+    public function generate_page_html() {
+        if ( ! current_user_can( 'manage_options' ) ) {
+            return;
+        }
+
+        $errors   = array();
+        $messages = array();
+        if ( class_exists( 'WPCOM_Legacy_Redirector' ) ) {
+            if ( isset( $_POST['from_url'] ) && isset( $_POST['redirect_to'] ) ) {
+                if (
+                    ! isset( $_POST['redirect_nonce_field'] )
+                    || ! wp_verify_nonce( $_POST['redirect_nonce_field'] )
+                ) {
+                    $errors[] = array(
+                        'label'   => __( 'Error', 'wpcom-legacy-redirector' ),
+                        'message' => __( 'Sorry, your nonce did not verify.', 'wpcom-legacy-redirector' ),
+                    );
+                } else {
+                    $from_url    = sanitize_text_field( $_POST['from_url'] );
+                    $redirect_to = sanitize_text_field( $_POST['redirect_to'] );
+                    if ( ! empty( $from_url ) && ! empty( $redirect_to ) && $from_url !== $redirect_to ) {
```

Comment:
That **checker** should be inside of [insert_legacy_redirect](https://github.com/Automattic/WPCOM-Legacy-Redirector/blob/master/wpcom-legacy-redirector.php#L75)? If you are inserting redirects from the command line you will face that issue.